app/org/sagebionetworks/bridge/models/upload/UploadFieldDefinition.java

```diff
@@ -0,0 +1,68 @@
+package org.sagebionetworks.bridge.models.upload;
+
+import javax.annotation.Nonnull;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.google.common.base.Strings;
+import org.springframework.validation.Errors;
+
+import org.sagebionetworks.bridge.dynamodb.DynamoUploadFieldDefinition;
+import org.sagebionetworks.bridge.models.BridgeEntity;
+import org.sagebionetworks.bridge.validators.Validate;
+
+/**
+ * This class represents a field definition for an upload schema. This could map to a top-level key-value pair in the
+ * raw JSON, or to a column in a Synapse table.
+ */
+@JsonDeserialize(as = DynamoUploadFieldDefinition.class)
+public interface UploadFieldDefinition extends BridgeEntity {
+    /** The field name. */
+    @Nonnull String getName();
+
+    /** True if the field is required to have data, false otherwise. */
+    boolean isRequired();
+
+    /**
+     * The field's type.
+     *
+     * @see org.sagebionetworks.bridge.models.upload.UploadFieldType
+     */
+    @Nonnull UploadFieldType getType();
+
+    /** This class validates the intrinsic properties of the field definition. */
+    static class Validator implements org.springframework.validation.Validator {
```

Comment:
Discussed this with Eric. I will move this to a separate class in the validators pacakge.